client/my-sites/exporter/export-card/index.jsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import { localize } from 'i18n-calypso';
+import flowRight from 'lodash/flowRight';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import SpinnerButton from './spinner-button';
+import FoldableCard from 'components/foldable-card';
+import analytics from 'lib/analytics';
+import Interval, { EVERY_SECOND } from 'lib/interval';
+import AdvancedSettings from './advanced-settings';
+import {
+	advancedSettingsFetch,
+	exportStatusFetch,
+	setPostType,
+	startExport,
+} from 'state/site-settings/exporter/actions';
+import {
+	shouldShowProgress,
+	getSelectedPostType,
+	isExporting,
+} from 'state/site-settings/exporter/selectors';
+
+class ExportCard extends Component {
+	componentWillMount() {
+		this.props.advancedSettingsFetch( this.props.siteId );
+	}
+
+	componentWillReceiveProps( newProps ) {
+		if ( newProps.siteId !== this.props.siteId ) {
+			this.props.advancedSettingsFetch( newProps.siteId );
+		}
+	}
+
+	render() {
+		const {
+			translate,
+			siteId,
+		} = this.props;
+		const exportAll = () => this.props.startExport( siteId );
+		const exportSelectedItems = () => this.props.startExport( siteId, { exportAll: false } );
+		const fetchStatus = () => this.props.exportStatusFetch( siteId );
```

Comment:
I feel like you can get rid of these three function variables and the need to get `this.props.siteId` if you add them to `mapDispatchToProps` and just pass them as `this.props.exportAll`, `this.props.exportSelectedItems` and `this.props.fetchStatus`. Then `mapDispatchToProps` would look something like:

```js
function dispatchStartExportEvent( dispatch, siteId, options ) {
    analytics.tracks.recordEvent( 'calypso_export_start_button_click', {
        scope: ( options && options.exportAll === false ) ? 'selected' : 'all'
    } );
    dispatch( startExport( siteId, options ) );
}

function mapDispatchToProps( dispatch, props ) {
    const { siteId } = props;
    const exportStatusFetch = flowRight( dispatch, exportStatusFetch );
    return {
        advancedSettingsFetch: flowRight( dispatch, advancedSettingsFetch ),
        setPostType: flowRight( dispatch, setPostType ),
        exportAll: () => dispatchStartExportEvent( dispatch, siteId ),
        exportSelectedItems: () => dispatchStartExportEvent( dispatch, siteId, { exportAll: false } ),
        fetchStatus: () => exportStatusFetch( siteId ),
    };
}
```